client/src/components/ProductCard.tsx

```diff
@@ -57,12 +56,14 @@ function ProductCard ({ product, bookmark, log }: Props) {
         } else {
           availabilityMsg = `While supplies last!\n(released ${weeks} weeks ago)`;
         }
+      } else if (product.end_date) {
+        setEndDate(product.end_date);
       }
```

Comment:
Will only display end date if product is already available and an end date exists.